functional-test/src/test/groovy/org/rundeck/tests/functional/api/scm/ScmPluginActionsSpec.groovy

```diff
@@ -0,0 +1,170 @@
+                    type: "String",
+                    values: null
+            ]
+            actionFields.fields[2].toMap() == [
+                    defaultValue: "true",
+                    description: "Check to push to the remote",
+                    name: "push",
+                    renderingOptions: [:],
+                    required: false,
+                    scope: null,
+                    title: "Push Remotely?",
+                    type: "Boolean",
+                    values: null
+            ]
+
+            actionFields.exportItems.size() == 1
+            actionFields.exportItems.first().toMap() == [
+                    deleted: false,
+                    itemId: "dummy-job-${jobUuid}.xml",
+                    job: [
+                            groupPath: null,
+                            jobId: "${jobUuid}",
+                            jobName: "dummy-job"
+                    ],
+                    status: "CREATE_NEEDED",
+                    originalId: null,
+                    renamed: false
+            ]
+        }
+    }
+
+    def "perform project scm action with success"(){
+        given:
+        String projectName = "${PROJECT_NAME}-autopush-${useAutoPush}"
+        setupProject(projectName, BASE_EXPORT_PROJECT_LOCATION)
+        GitScmApiClient scmClient = new GitScmApiClient(clientProvider).forIntegration(integration).forProject(projectName)
+        String jobUuid = getJobUuid(projectName, 'dummy-job')
+        String keyStoragePath = 'keys/scm.password'
+
+        ScmActionPerformRequest actionRequest = new ScmActionPerformRequest([
+                input: [ message : 'Commit msg example', push: useAutoPush.toString()],
+                jobs: [ jobUuid ]
+        ])
+
+
+        expect:
+        if(useAutoPush)
+            remoteRepo.storeRepoPassInRundeck(new KeyStorageApiClient(clientProvider), keyStoragePath).successful
+        scmClient.callSetupIntegration(GitExportSetupRequest.defaultRequest().forProject(projectName).withRepo(remoteRepo)).response?.success
+
+        when:
+        SetupIntegrationResponse performActionResult = scmClient.callPerformAction(actionId, actionRequest).response
+
+        then:
+        IntegrationStatusResponse finalScmStatus = scmClient.callGetIntegrationStatus().response
+        verifyAll {
+            performActionResult.success
+            performActionResult.message == "SCM ${integration} Action was Successful: ${actionId}"
+            finalScmStatus.actions == expectedFinalScmActions
+            finalScmStatus.synchState == expectedFinalSynchState
+        }
+
+        where:
+        integration | actionId          | useAutoPush | expectedFinalSynchState | expectedFinalScmActions
+        'export'    | 'project-commit'  | false       | 'EXPORT_NEEDED'         | ['project-push']
+        'export'    | 'project-commit'  | true        | 'CLEAN'                 | null
+    }
+
+
+    String getJobUuid(String projectName, String jobName){
```

Comment:
this could go in "JobUtils"

Reply:
Replaced with `JobUtils.jobImportFile` which returns an object that contains the jobuuid